src/evox/problems/numerical/lsmop.py

```diff
@@ -0,0 +1,755 @@
+
+    def evaluate(self, state: chex.PyTreeDef, X: chex.Array):
+        chex.assert_type(X, float)
+        chex.assert_shape(X, (None, self.n))
+        return jax.jit(jax.vmap(self._lsmop))(X), state
+
+    def pf(self, state: chex.PyTreeDef):
+        f = UniformSampling(self.ref_num * self.m, self.m)()[0] / 2
+        return f, state
+
+    @staticmethod
+    def _Sphere(x):
+        """get the sum of squares of each row in matrix x"""
+        return jnp.sum(x**2, axis=1, keepdims=True)
+
+    @staticmethod
+    def _Giewank(x):
+        f = (
+            jnp.sum(x**2, axis=1, keepdims=True) / 4000
+            - jnp.prod(
+                jnp.cos(
+                    x
+                    / jnp.tile(
+                        jnp.sqrt(jnp.arange(1, jnp.shape(x)[1] + 1)),
+                        (jnp.shape(x)[0], 1),
+                    )
+                ),
+                axis=1,
+                keepdims=True,
+            )
+            + 1
+        )
+        return f
+
+    @staticmethod
+    def _Schwefel(x):
+        return jnp.max(jnp.abs(x), keepdims=True, axis=1)
+
+    @staticmethod
+    def _Rastrigin(x):
+        f = jnp.sum(x**2 - 10 * jnp.cos(2 * jnp.pi * x) + 10, axis=1, keepdims=True)
+        return f
+
+    @staticmethod
+    def _Rosenbrock(x):
+        f = jnp.sum(
+            100 * ((x[:, : jnp.shape(x)[1] - 1]) ** 2 - x[:, 1 : jnp.shape(x)[1]]) ** 2
+            + (x[:, : jnp.shape(x)[1] - 1] - 1) ** 2,
+            axis=1,
+            keepdims=True,
+        )
+        return f
+
+    @staticmethod
+    def _Ackley(x):
+        # np.seterr(divide='ignore', invalid='ignore')
+        f = (
+            20
+            - 20
+            * jnp.exp(
+                -0.2
+                * jnp.sqrt(jnp.sum(x**2, axis=1, keepdims=True) / jnp.shape(x)[1])
+            )
+            - jnp.exp(
+                jnp.sum(jnp.cos(2 * jnp.pi * x), axis=1, keepdims=True)
+                / jnp.shape(x)[1]
+            )
+            + jnp.exp(1)
+        )
+        if jnp.isnan(f[0]):
+            f = jnp.zeros([jnp.shape(f)[0], jnp.shape(f)[1]])
+        return f
+
+    @staticmethod
+    def _Griewank(x):
+        f = (
+            jnp.sum(x**2, axis=1, keepdims=True) / 4000
+            - jnp.prod(
+                jnp.cos(
+                    x
+                    / jnp.tile(
+                        jnp.sqrt(jnp.arange(1, jnp.shape(x)[1] + 1)),
+                        (jnp.shape(x)[0], 1),
+                    )
+                ),
+                axis=1,
+                keepdims=True,
+            )
+            + 1
+        )
+        return f
+
+
+class LSMOP1(LSMOP):
+    def __init__(self, d=None, m=None, ref_num=1000):
+        super().__init__(d, m, ref_num)
+        if m is None:
+            self.m = 3
+        else:
+            self.m = m
+        if d is None:
+            self.d = self.m + 4
+        else:
+            self.d = d
+
+    def evaluate(self, state: chex.PyTreeDef, X: chex.Array):
```

Comment:
Don't use chex for type hint, we are trying to move away from chex, since JAX is building it's own types.